src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/MemoryCache.cs

```diff
@@ -6,29 +6,32 @@
 using System.Collections.Generic;
 using System.Linq;
 
-namespace Microsoft.AspNetCore.Razor.LanguageServer
+namespace Microsoft.CodeAnalysis.Razor
 {
     // We've created our own MemoryCache here, ideally we would use the one in Microsoft.Extensions.Caching.Memory,
     // but until we update O# that causes an Assembly load problem.
-    internal class MemoryCache<TResult>
+    internal class MemoryCache<TKey, TValue>
     {
-        protected virtual int SizeLimit { get; } = 50;
+        private const int DefaultSizeLimit = 50;
 
-        protected IDictionary<string, CacheEntry> _dict;
+        protected IDictionary<TKey, CacheEntry> _dict;
 
-        public MemoryCache()
+        public MemoryCache(int sizeLimit = DefaultSizeLimit)
         {
-            _dict = new ConcurrentDictionary<string, CacheEntry>(concurrencyLevel: 2, capacity: SizeLimit);
+            SizeLimit = sizeLimit;
+            _dict = new ConcurrentDictionary<TKey, CacheEntry>(concurrencyLevel: 2, capacity: SizeLimit);
         }
 
-        public bool TryGetValue(string key, out TResult result)
+        private int SizeLimit { get; }
```

Comment:
Any reason to have it as a private property instead of just a field?

Reply:
Field should be fine too, I'll address this in https://github.com/dotnet/aspnetcore-tooling/pull/2307